MekHQ/src/mekhq/campaign/mission/atb/scenario/PirateFreeForAllBuiltInScenario.java

```diff
@@ -77,32 +81,35 @@ public boolean canRerollMapSize() {
 
     @Override
     public void setExtraMissionForces(Campaign campaign, ArrayList<Entity> allyEntities,
-            ArrayList<Entity> enemyEntities) {
+                                      ArrayList<Entity> enemyEntities) {
         setStart(Board.START_CENTER);
 
+        final AtBContract contract = getContract(campaign);
+
         for (int i = 0; i < 4; i++) {
-            getAlliesPlayer()
-                    .add(getEntity(getContract(campaign).getEmployerCode(), getContract(campaign).getAllySkill(),
-                            getContract(campaign).getAllyQuality(), UnitType.MEK, UnitMarket.getRandomAeroWeight(), // max
-                                                                                                                    // heavy
-                            campaign));
+            int weightClass;
+            do {
+                weightClass = AtBMonthlyUnitMarket.getRandomWeight(campaign, UnitType.MEK, contract.getEmployerFaction());
+            } while (weightClass == EntityWeightClass.WEIGHT_ASSAULT);
```

Comment:
If this is maximum of Heavy, should we ignore superheavies too?
```suggestion
            } while (weightClass >= EntityWeightClass.WEIGHT_ASSAULT);
```

Reply:
AtB weight generation is Light to Assault only.